src/libertem/win_tweaks.py

```diff
@@ -47,10 +50,13 @@ def set_console_mode(mode, stream=sys.stdin):
 
 
 def disable_quickedit():
-    mode = get_console_mode()
-    mode &= ~ENABLE_QUICK_EDIT
-    mode |= ENABLE_EXTENDED_FLAGS
-    set_console_mode(mode)
+    try:
+        mode = get_console_mode()
+        mode &= ~ENABLE_QUICK_EDIT
+        mode |= ENABLE_EXTENDED_FLAGS
+        set_console_mode(mode)
+    except TypeError:
```

Comment:
Is `TypeError` here the right exception type to catch? Not `ctypes.WinError`? Where does the `TypeError` come from?

Reply:
TypeError was there in the original file. Also, the issue here was incompatibility of types.

Reply:
Sorry, I don't understand...

Reply:
`mode` is supposed to be `int`. However, if we get an error, `get_console_mode()` returns `NoneType` and we get a `TypeError` like this:

`line 51, in disable_quickedit mode &= ~ENABLE_QUICK_EDIT `               
`TypeError: unsupported operand type(s) for &=: 'NoneType' and 'int'`

Reply:
> However, if we get an error, `get_console_mode()` returns `NoneType`

Hmm, but if there is an error in `get_console_mode`, it throws an exception, which means it doesn't return anything...

Reply:
If we use `except` keyword, we cannot put `ctypes.WinError` as it doesn't inherit from `BaseException` class. However, if we use `raise ctypes.WinError()` we get 
` File "c:\users\saswati\venv\libertem\src\libertem\win_tweaks.py", line 45, in get_console_mode`
    `raise ctypes.WinError(err)`
`OSError: [WinError 0] The handle is invalid.`

Thus, back to square one.

Reply:
> back to square one

Have a close look at the traceback again: apparently, the exception type is `OSError`, not `WinError`. Try to catch `OSError` and see if that fixes things.

Reply:
Right, `OSError` works just fine.